pkg/k8splus/pods.go

```diff
@@ -2,26 +2,52 @@ package k8splus
 
 import (
 	"context"
+	"sync"
 
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	kv1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
+// ExtendedPodInterface is the interface for the Pods method
+// This interface extends the PodInterface to include the CreateBatch method
 type ExtendedPodInterface interface {
 	kv1.PodInterface
-	CreateBatch(ctx context.Context, pods []*v1.Pod, opts metav1.CreateOptions) ([]*v1.Pod, error)
+	CreateBatchPods(ctx context.Context, pods []*v1.Pod, opts metav1.CreateOptions) ([]*v1.Pod, error)
 }
 
 // pods implements ExtendedPodInterface
 type pods struct {
 	corev1.PodInterface
 }
 
-func (c *pods) CreateBatch(ctx context.Context, pods []*v1.Pod, opts metav1.CreateOptions) ([]*v1.Pod, error) {
-	var createdPods []*v1.Pod
-	// bla bla bla
+// CreateBatchPods creates multiple pods in a single request
+// It's a wrapper around the Create method and adds the CreateBatchPods method
+func (c *pods) CreateBatchPods(ctx context.Context, pods []*v1.Pod, opts metav1.CreateOptions) ([]*v1.Pod, error) {
+	createdPods := make([]*v1.Pod, len(pods))
+	errors := make(chan error, len(pods))
+	var wg sync.WaitGroup
+
+	for i, pod := range pods {
+		wg.Add(1)
+		go func(pod *v1.Pod, index int) {
+			defer wg.Done()
+			createdPod, err := c.PodInterface.Create(ctx, pod, opts)
```

Comment:
Nice one; however, this `Create` is calling the API under the hood so we are not batching here. 
We need to create a batch request to the associated k8s endpoint and make it all in one request.
something similar to this one for example:

```go
	podList := &v1.PodList{Items: []v1.Pod{}}
	for _, pod := range pods {
		podList.Items = append(podList.Items, *pod)
	}

	body, err := json.Marshal(podList)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal pods: %w", err)
	}

	req := p.client.Post().
		Namespace(p.ns).
		Resource("pods").
		VersionedParams(&opts, scheme.ParameterCodec).
		Body(body).
		Do(ctx)

	result := &v1.PodList{}
	err = req.Into(result)
	if err != nil {
		return nil, fmt.Errorf("failed to create pods: %w", err)
	}

	createdPods := []*v1.Pod{}
	for _, pod := range result.Items {
		createdPods = append(createdPods, &pod)
	}
```

Not sure if it works, did not test it.